pkg/base/helm.go

```diff
@@ -74,6 +74,32 @@ func RenderHelm(u *upstreamtypes.Upstream, renderOptions *RenderOptions) (*Base,
 		return nil, errors.Errorf("unknown helmVersion %s", renderOptions.HelmVersion)
 	}
 
+	// add namespace if it's defined in the spec and not already present in the manifests
+	if renderOptions.Namespace != "" {
+		for i, v := range rendered {
+			var manifest map[string]interface{}
+			err := yaml.Unmarshal(v.Content, &manifest)
+			if err != nil {
+				return nil, errors.Wrap(err, "failed to unmarshal helm manifest")
+			}
+
+			if metadata, ok := manifest["metadata"].(map[interface{}]interface{}); ok {
+				if _, ok := metadata["namespace"]; !ok {
+					metadata["namespace"] = renderOptions.Namespace
+				} else {
+					continue // don't bother overwriting the yaml if a namespace already exists
+				}
+			}
+
+			b := new(bytes.Buffer)
+			err = yaml.NewEncoder(b).Encode(manifest)
```

Comment:
could use yaml.Marshal instead for consistency sake

Reply:
done